// carbon-components-react/lib/components/PaginationNav/next/PaginationNav.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports._PaginationNav = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = _interopRequireDefault(require("../"));

/**
 * Copyright IBM Corp. 2020
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var props = function props() {
  return {
    loop: false,
    page: 0,
    totalItems: 10,
    itemsShown: 10
  };
};

var _default = {
  title: 'Components/PaginationNav',
  decorators: [function (story) {
    return /*#__PURE__*/_react.default.createElement("div", {
      style: {
        width: '800px'
      }
    }, story());
  }],
  parameters: {
    component: _.default
  }
};
exports.default = _default;

var _PaginationNav = function _PaginationNav() {
  return /*#__PURE__*/_react.default.createElement(_.default, props());
};

exports._PaginationNav = _PaginationNav;
