// carbon-components-react/es/components/ComboBox/next/ComboBox.stories.js
/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from 'react';
import ComboBox from '../';
import { Layer } from '../../Layer';
var items = [{
  id: 'option-0',
  text: 'An example option that is really long to show what should be done to handle long text'
}, {
  id: 'option-1',
  text: 'Option 1'
}, {
  id: 'option-2',
  text: 'Option 2'
}, {
  id: 'option-3',
  text: 'Option 3 - a disabled item',
  disabled: true
}, {
  id: 'option-4',
  text: 'Option 4'
}, {
  id: 'option-5',
  text: 'Option 5'
}];
export default {
  title: 'Components/ComboBox',
  component: ComboBox
};
export var Combobox = function Combobox() {
  return /*#__PURE__*/React.createElement("div", {
    style: {
      width: 300
    }
  }, /*#__PURE__*/React.createElement(ComboBox, {
    onChange: function onChange() {},
    id: "carbon-combobox",
    items: items,
    itemToString: function itemToString(item) {
      return item ? item.text : '';
    },
    placeholder: "Filter...",
    titleText: "ComboBox title",
    helperText: "Combobox helper text"
  }));
};
export var withLayer = function withLayer() {
  return /*#__PURE__*/React.createElement("div", {
    style: {
      width: 300
    }
  }, /*#__PURE__*/React.createElement(ComboBox, {
    onChange: function onChange() {},
    id: "carbon-combobox",
    items: items,
    itemToString: function itemToString(item) {
      return item ? item.text : '';
    },
    placeholder: "Filter...",
    titleText: "First Layer",
    helperText: "Combobox helper text"
  }), /*#__PURE__*/React.createElement(Layer, null, /*#__PURE__*/React.createElement(ComboBox, {
    onChange: function onChange() {},
    id: "carbon-combobox",
    items: items,
    itemToString: function itemToString(item) {
      return item ? item.text : '';
    },
    placeholder: "Filter...",
    titleText: "Second Layer",
    helperText: "Combobox helper text"
  }), /*#__PURE__*/React.createElement(Layer, null, /*#__PURE__*/React.createElement(ComboBox, {
    onChange: function onChange() {},
    id: "carbon-combobox",
    items: items,
    itemToString: function itemToString(item) {
      return item ? item.text : '';
    },
    placeholder: "Filter...",
    titleText: "Third Layer",
    helperText: "Combobox helper text"
  }))));
};
